### DISCUSSION TITLE
Button in WYSIWYG editor to insert a soft-hyphen (&shy;)

### DISCUSSION THREAD

**Discussion by OttoKerner** [GetPublii/Publii#1636] [Ideas]
When writing in a language which tends to have long words (like German for example), it can help to insert soft hyphens into longer words. (HTML entity name `&shy;`)
The browser will use these as possible locations for word wrapping, particularly helpful on small screens and narrow text columns.

The editor is supporting them just fine when copy&pasting text from existing web pages. They continue to work in the published pages.
However, as far as I have found out, the only way to insert them in Publii is to paste them in from an external source, or by using some code number input method (e.g. `Alt`+`173` on Windows).

If there was a button to insert a soft hypen in the editor, life would be so much easier.
Bonus points for a visible hint in the text preview, where a soft hyphen has been placed.

**Comment by candidexmedia** [comment]
Give this gist a read: https://gist.github.com/candidexmedia/367711a80274b4c005fd5983df97d20c

A template plugin might work?

**Reply by OttoKerner** [reply]
Thank you for the tip! Unfortunately, I saw this at the bottom: 

> Also note that Templates are a deprecated TinyMCE plugin, and will become a paid TinyMCE plugin in version 7 ([see old forum discussion](https://forum.getpublii.com/topic/tinymce-wysiwyg-editor-adding-more-blocks-html-tags-to-the-dropdown/#post-11739)).

Since this is only for a small project, I'm not sure it is worth buying the TinyMCE license.

**Reply by candidexmedia** [reply]
> Since this is only for a small project, I'm not sure it is worth buying the TinyMCE license.

TinyMCE 6 is intergrated into Publii. As such, it would be up to the Publii developpers to decide whether they wanted to pay for that plugin, and not you. In fact, they could also opt to keep the current TinyMCE version and not upgrade since it serves its purpose.

The discussion I linked to in my gist talks about that.
